### ISSUE TITLE
error: type checker + error: result of statement 'double %1 = sitoptr 1' is not a pointer

### ISSUE THREAD

**Issue by practicalswift** [NASA-SW-VnV/ikos#58]
```
$ cat testcase.c
struct {
  char a[0];
  int *b;
  double c
} d = {{}, 1};
main() {}
$ ikos testcase.c
[*] Compiling testcase.c
…
[*] Running ikos preprocessor
[*] Running ikos analyzer
[*] Translating LLVM bitcode to AR
error: result of statement 'double %1 = sitoptr 1' is not a pointer
ikos-analyzer: /tmp/ikos-tIUY4M/testcase.pp.bc: error: type checker
ikos: error: a run-time error occurred
```

**Comment by arthaud** [comment]
In debug mode:
```
$ ikos test.c
[*] Compiling test.c
…
[*] Running ikos preprocessor
[*] Running ikos analyzer
[!] ikos was compiled in debug mode, the analysis might be slow
[*] Translating LLVM bitcode to AR
Assertion failed: (cst->getNumOperands() == type->num_fields()), function translate_constant_struct, file /Users/marthaud/ikos/ikos-git/frontend/llvm/src/import/constant.cpp, line 262.
0  ikos-analyzer            0x0000000101473d92 llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 37
1  ikos-analyzer            0x0000000101474196 SignalHandler(int) + 200
2  libsystem_platform.dylib 0x00007fff7d2c4f5a _sigtramp + 26
3  libsystem_platform.dylib 0x00007fec7b2005c8 _sigtramp + 4260607624
4  libsystem_c.dylib        0x00007fff7d0621ae abort + 127
5  libsystem_c.dylib        0x00007fff7d02a1ac basename_r + 0
6  ikos-analyzer            0x0000000101088c71 ikos::frontend::import::ConstantImporter::translate_constant_struct(llvm::ConstantStruct*, ikos::ar::StructType*, ikos::ar::BasicBlock*, llvm::SmallVectorImpl<ikos::frontend::import::ConstantImporter::ConstantExpression>&) + 177
7  ikos-analyzer            0x0000000101085d8b ikos::frontend::import::ConstantImporter::translate_constant(llvm::Constant*, ikos::ar::Type*, ikos::ar::BasicBlock*, llvm::SmallVectorImpl<ikos::frontend::import::ConstantImporter::ConstantExpression>&) + 1563
8  ikos-analyzer            0x0000000101084e83 ikos::frontend::import::ConstantImporter::translate_constant(llvm::Constant*, ikos::ar::Type*, ikos::ar::BasicBlock*) + 115
9  ikos-analyzer            0x000000010107e7e5 ikos::frontend::import::BundleImporter::translate_global_variable_initializer(llvm::GlobalVariable*) + 309
10 ikos-analyzer            0x00000001010d24a2 ikos::frontend::import::Importer::import(llvm::Module&, ikos::ar::Flags<ikos::frontend::import::Importer::ImportOption>) + 2402
11 ikos-analyzer            0x000000010096dc03 main + 5555
12 libdyld.dylib            0x00007fff7cfb6015 start + 1
13 libdyld.dylib            0x000000000000000f start + 2198118395
Stack dump:
0.	Program arguments: /Users/marthaud/ikos/ikos-install/bin/ikos-analyzer -a=sound,shc,nullity,pcmp,dfa,dca,fca,prover,dbz,boa,sio -d=interval -entry-points=main -globals-init=skip-big-arrays -prec=mem -proc=inter -allow-dbg-mismatch -display-checks=no -display-inv=no -color=auto -log=info /var/folders/9g/hvg_xszs3q79k8tvtqhm_q5s5y3dxb/T/ikos-ADHpBq/test.pp.bc -o output.db
ikos: error: exited with signal SIGIOT
```

PS: You should consider building ikos in debug mode if you are doing fuzzing ;)

**Comment by arthaud** [comment]
This is tracked by #67

**Comment by arthaud** [comment]
This should be fixed by 1c12398390fdaba4797ea384ec9ec1172c2d9c7b. Feel free to reopen if the issue is still here.

**Comment by practicalswift** [comment]
Confirming that I'm not longer encountering this issue. Thanks!
